### DISCUSSION TITLE
[Feature] Android: Enable Immich to become default photo viewer app

### DISCUSSION THREAD

**Discussion by SlavikCA** [immich-app/immich#8116] [Feature Request]
### The feature

I decided to switch from Google Photos app to Immich, so I uninstalled Google Photos from my Pixel 6 device.

However, now other apps, such as "Camera", "Voice" can't view photos (see screenshots below).

Can Immich app be registered / configured in the Android system to be "photo viewer app"?

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by SlavikCA** [comment]
![Camera](https://github.com/immich-app/immich/assets/6293679/e8971229-ac1f-4ab0-aa9f-2d1491f7b6e7)

![Voice](https://github.com/immich-app/immich/assets/6293679/ee59ac4f-878c-4b66-b65e-c7201aa9bd78)

**Reply by alextran1502** [reply]
We just add a permission in Android for this. Can you help double check in Immich app setting on Android?

**Reply by Atemu** [reply]
I don't know how it works in Flutter but you must tell the android system that Immich can receive certain action intents like `com.android.camera.action.REVIEW` for opening a taken picture from the camera.

See i.e. https://github.com/FossifyOrg/Gallery/blob/d19f8156b9385fe0794cf086c8672c26e97c90b0/app/src/main/AndroidManifest.xml#L111-L119 for reference (and then obviously handle those internally somewhere)

Again, I don't know how this translates into Flutter.

The media permission is just for generally being allowed to manage the media files.

**Comment by bo0tzz** [comment]
Duplicate of #1605

**Reply by gaaat98** [reply]
Hi! Sorry for the reply to this old comment but I don't think it's a duplicate of that, at least not the part regarding the behaviour in the Google Camera App, where the prompt to install Photos appears. That prompt is still present after upgrading to Immich v3 and setting Immich as default image viewer. I believe that something along the lines of this: https://github.com/nermolov/google-pixel-camera-redirect should address that problem, but I am not an android dev so I don't know if it can be implemented also in Immich. Nevertheless it would be a great addition for people wanting to completely remove Google Photos from their devices (like me 😄)

**Comment by chancedonmillion** [comment]
I too would like this feature.

**Comment by pkoevesdi** [comment]
For anyone landing here from a camera app: #26109 (shipped in v3.0.0) registers Immich for `android.intent.action.VIEW` only. The `com.android.camera.action.REVIEW` part mentioned in the comment above is still missing, so a camera app's "last shot" button never offers Immich. Follow-up request: #31684
